mmv1/templates/terraform/constants/artifact_registry_repository.go.erb

```diff
@@ -0,0 +1,34 @@
+<%# The license inside this block applies to this file
+  # Copyright 2023 Google Inc.
+  # Licensed under the Apache License, Version 2.0 (the "License");
+  # you may not use this file except in compliance with the License.
+  # You may obtain a copy of the License at
+  #
+  #     http://www.apache.org/licenses/LICENSE-2.0
+  #
+  # Unless required by applicable law or agreed to in writing, software
+  # distributed under the License is distributed on an "AS IS" BASIS,
+  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  # See the License for the specific language governing permissions and
+  # limitations under the License.
+-%>
+func upstreamPoliciesDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+  o, n := d.GetChange("virtual_repository_config.0.upstream_policies")
+  oldPolicies, ok := o.([]any)
+  if !ok {
+    return false
+  }
+  newPolicies, ok := n.([]any)
+  if !ok {
+    return false
+  }
+  hashFunc := func(val any) int {
+    x := val.(map[string]any)
+    hashStr := fmt.Sprintf("[id:%v priority:%v repository:%v]", x["id"], x["priority"], x["repository"])
```

Comment:
Is there any chance we could see a panic if val isn't a map?

Reply:
It is possible if the returned field doesn't fit the expected schema, somehow. I updated it to avoid panicking on a bad cast.